### ISSUE TITLE
Directories not displaying subdirectories in cmder

### ISSUE THREAD

**Issue by fstylermiller** [scrooloose/nerdtree#738]
_Running windows 7 using cmder. I have my vimrc saved in `C:\Users\$HOME\.vimrc` and my plugins are installed in `C:\Users\$HOME\.vim`_

>Operating System: Windows 7
>Vim version: VIM - Vi IMproved 8.0
>NERDTree version: 5782b22
>NERDTree settings applied in your vimrc:
>   ```
>let g:NERDTreeWinPos = "left" 
>let g:NERDTreeWinSize = 21
>let g:NERDSpaceDelims = 1     
>   ```

**Process:**

>1. Open cmder
>2. Open a directory using vim.

**Current Result:**

>- Directories displayed do not open subdirectories when navigating using `j`/`k` and hitting `enter` or using any other form of "opening" a directory. Such as `o`, `i`, or `C`.
>- Files in the current directory open as expected.
>- I can navigate up a dir by selecting `.. (up a dir)`

**Expected Result:**

>Subdirectories should be displayed

---
**Screenshot(s):**

![3dirs and a txt file, hitting enter does nothing](http://i.imgur.com/kzYM9fW.png)

**Comment by lifecrisis** [comment]
Are you using the `'shellslash'` setting? Try turning it on in your `.vimrc` and see what happens.

**Comment by fstylermiller** [comment]
As to what shell I'm using, just MS DOS. I could be wrong but I believe that Cmder provides it's own shell.
Setting shellslash to true/false made no difference unfortunately.

**Comment by lifecrisis** [comment]
Is the directory path at the root of the tree correct? Is there really a `C:\c\` directory?

**Comment by fstylermiller** [comment]
There is not. No idea why that may be there.

**Comment by lifecrisis** [comment]
Interesting, so the problem is that you can't load any new directories after the first instance loads.

Can you try to use Gvim on your Windows 7 machine? Does the NERDTree experience the same problems there? Try your best to recreate your problem exactly.

**Comment by PhilRunninger** [comment]
It looks like he's using a bash shell in Windows, in which the `C:\` drive is denoted `/c/`. The NERDTree root is the `test` subdirectory of his home folder.

**Comment by fstylermiller** [comment]
Cmder is providing the bash-like shell.

**Comment by lifecrisis** [comment]
My advice at this point is to try switching up the underlying shell for cmder.  If you can configure cmder to use Powershell or cmd.exe, then all the better.  This is truly a bizarre issue.

**Comment by fstylermiller** [comment]
I switched to both of those shells, and still had the same issue. Cmder is weird.

**Comment by fstylermiller** [comment]
Also I noticed Cmder is using powershell as the default. It just switches in secret on startup.
![Powershell](http://i.imgur.com/GwvorQP.png)

**Comment by lifecrisis** [comment]
Can you run

```viml
:echo getcwd
```

in the Vim command line and tell me what prints out?

**Comment by fstylermiller** [comment]
```
:echo getcwd()
```

Properly returns the current path

```
/c/Users/MILLETY/test
```

Nothing works in gvim, I can't seem to get vundle to work with gvim as this computer is a bit of a corporate mess. I'll look into gvim a bit more.

**Comment by lifecrisis** [comment]
Given the situation, you may want to report this issue to the cmder maintainers. Just link them here and see what they say.

If cmder changes the representation of pathnames generally (i.e., from `C:\Users\` to something like `/c/Users/`), I'm not sure there's anything we can do to solve the problem other than a bunch of code and a new option for users running cmder.

**Comment by fstylermiller** [comment]
I'll let them know. I'm not sure how your code-base is written, but I find it weird that I can navigate up a directory, but not down through any. That leads me to believe that it's not an issue with `/c/`. But once again I'm not familiar with your code-base. ¯\\_(ツ)_/¯

**Comment by lifecrisis** [comment]
The NERDTree operates on Windows by adding and removing the drive letter in several places. It does this by using the normal `C:\` format.

This new convention for writing Windows pathnames makes the behavior of the NERDTree totally unpredictable. If pathnames in Windows no longer have the expected format, all bets are off.

**Comment by PhilRunninger** [comment]
@fstylermiller, Could you try this change? I noticed the `~V` in your screenshot where the expand/collapse glyphs would be. I suspect your font isn't able to show the default Unicode characters, and that is causing your problems. Either try another font that shows the triangles, or override them in your vimrc with these statements:
```
let NERDTreeDirArrowExpandable = '+'
let NERDTreeDirArrowCollapsible = '~'
```
Also, I think NERDTree may be confused by your shell, thinking your not a on Windows machine. Run these commands and report back what they return.
```
:echo has("win16")
:echo has("win32")
:echo has("win64")
:echo has("win32unix")
```
I'd merged a pull request, #626, to add `has("win32unix")` to the expression in https://github.com/scrooloose/nerdtree/blob/master/autoload/nerdtree.vim#L153-L156, but it was reported to have caused other issues, so I reveted it.

**Comment by fstylermiller** [comment]
Oh wow. Setting those two variables fixed the issue.

The results of those commands were:
```
:echo has("win16") -------> 0
:echo has("win32") -------> 0
:echo has("win64") -------> 0
:echo has("win32unix") ---> 1
```

Thank's so much for the fix!
